docs/advanced/routing/external-routing.md

```diff
@@ -0,0 +1,100 @@
+# External Routing
+
+It is possible to resolve routes dynamically based on platform data. Daffodil enables this behavior with [`@daffodil/external-router`](../../../libs/external-router/README.md).
+
+## Getting Started
+
+### Configuration
+
+`@daffodil/external-router` needs information about how to construct the route. This info can be provided as an argument to [`DaffExternalRouterModule#forRoot`](../../../libs/external-router/src/external-router.module.ts).
+
+#### Insertion Strategy
+
+If routes need to be added at specific points in the router config, the `@daffodil/external-router` configuration exposes [`DaffTypeRoutePair#insertionStrategy`](../../../libs/external-router/src/external-router.module.ts). The insertion strategy can be used to customize the behavior for adding routes.
+
+> See [`@daffodil/external-router` configuration](../../../libs/external-router/guides/configuration.md).
+
+### Route Capture
+
+Guard your wildcard route with [`DaffExternalRouterExistenceGuard`](../../../libs/external-router/routing/src/guard/existence.guard.ts). This will attempt to resolve routes with your platform of choice. If a particular route is resolved by the platform then the guard will invoke the [`DaffExternalRouter`](../../../libs/external-router/src/router/router.service.ts) service and add it to the local route configuration.
+
+For the remainder of the session, subsequent navigations to that path will bypass the wildcard route and be handled directly by the added route.
+
+## Example
+<!-- TODO(griest024): change import path for product routing subpackage -->
+`app.module.ts`
```

Comment:
Could you add a short description about how you're configuring the `DaffExternalRouterModule` and everything you're passing in? Also, it isn't super easy to see what you're configuring because of how things are indented and all of the brackets and curlies. I think it would be a little more clear like this:
```
[
    DaffExternalRouterModule.forRoot(
       { failedResolutionPath: 'not-found', }, 
       [{
           type: 'product',
           route: {
              component: MyProductPageComponent,
              ..., // additional route config, but without path
              resolve: {
                 product: DaffProductPageUriResolver
              }
          }
      }]
   ),
]
```
or something like that, so it's easy to tell that is all nested within the `forRoot` method

Reply:
changed, lmk if you think it needs something else

Reply:
One last thing that I think will be helpful is to add a concrete example of one workflow. I imagine it something like this:

For example the resolvable route, i.e. a uri the backend platform will recognize and resolve to data, `product/my-product-name.html` would follow this workflow, given the configurations below:
1. **The User Initializes a Navigation:** A user triggers Angular to route to the `product/my-product-name.html`.
2. **The Route is Processed By Angular Router:** This route is not defined statically in the angular configuration of the app, so it gets caught by the wildcard (`**`) route and gets directed to the `DaffExternalRouterExistenceGuard`.
3. **The DaffExternalRouterExistenceGuard:** Where the magic happens.
    a. The `DaffExternalRouterExistenceGuard` verifies with the backend platform (through the `DaffExternalRouterDriver`) that `product/my-product-name.html` is indeed a resolvable route and that this particular route resolves to a product.
    b. Upon successful verification, the `DaffExternalRouterExistenceGuard` dynamically inserts (through the `DaffExternalRouter`) the `product/my-product-name.html` route into the list of Angular routes using the `productRouteConfig` defined below as configurations.
    c. The angular router is instructed to reroute to the same route now that `product/my-product-name.html` will be recognized by the Angular Router.
4. **The Route is Processed By Angular Router Again:** This time, the `product/my-product-name.html` route will be recognized and is instructed to use the `DaffProductPageUriResolver` to resolve the product data.
5. Once the data loads, it will be rendered with the `MyProductPageComponent`.

Reply:
nice, added

Reply:
I don't think markdown supports this but I could be wrong.

Reply:
that's unlucky